### PR TITLE
50: Add List component

### LINKED ISSUE #50: Add List component
- Simple list
- List with icon
- List with dividers
- List with icons and dividers
- Inline list
- Middot-seperated inline list
- Split list

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
# Done
- Added [List](https://docs.vanillaframework.io/en/patterns/lists) component and ListItem sub-component
- Includes Default, Ticked, Divided, Inline, Middot and Split options
- Added stories to Storybook
- Added snapshot tests
- **Note:** I started building it out so that 'stepped' and 'stepped-detailed' could also be modifiers, but I think it would work better as a separate component as it's wired a bit differently to the rest of the list types

# QA
- Pull code
- Run `yarn clean` and `yarn serve` and navigate to http://localhost:6006/
- Run `yarn lint` and `yarn test` to ensure there are no linting or testing errors
- Verify the List component in Storybook

# Fixes
Fixes #50

### CODE REVIEW

File: src/lib/components/List/List.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import 'vanilla-framework/scss/base';
+@import 'vanilla-framework/scss/patterns_lists';
+@include vf-b-typography;
+@include vf-p-lists;
+
+.p-inline-list--middot .p-inline-list__item::after {
+  right: -0.8rem;
```
Comment (anthonydillon) [inline src/lib/components/List/List.scss]: Is this a fix?

File: src/lib/components/List/List.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+}
+
+List.defaultProps = {
+  modifier: null,
+};
+
+List.propTypes = {
+  children: PropTypes.node.isRequired,
+  modifier: PropTypes.oneOf(['divided', 'inline', 'middot', 'split']),
```
Comment (anthonydillon) [inline src/lib/components/List/List.js]: null or default for p-list?

**Review by anthonydillon — APPROVED** [review]
Looks good, few small questions.

File: src/lib/components/List/List.scss (hunk 1 above)
Comment (barrymcgee) [reply src/lib/components/List/List.scss]: Just a reminder - this project is an implementation of Vanilla Framework, not an augmentation of it. Vanilal famowrk should remain the single source of truth for all styles.

Comment (anthonydillon) [reply src/lib/components/List/List.scss]: I agree, I think this repo should have no CSS in it.

File: src/lib/components/List/List.js (hunk 2 above)
Comment (Caleb-Ellis) [reply src/lib/components/List/List.js]: @anthonydillon Is it necessary if 'modifier' is not a required prop?

File: src/lib/components/List/List.scss (hunk 1 above)
Comment (Caleb-Ellis) [reply src/lib/components/List/List.scss]: @anthonydillon Got it - I'll remove the css. I'm not sure why the dots on the middot list are showing up out of place though.

Comment (anthonydillon) [reply src/lib/components/List/List.scss]: Not sure, it should be a Vanilla issue. Could you file an issue please?

File: src/lib/components/List/List.js (hunk 2 above)
Comment (anthonydillon) [reply src/lib/components/List/List.js]: Ah, I see it's not required. Thanks.

**Review by anthonydillon — APPROVED** [review]
Thanks :+1:

**Pull request merged**